DataFormats/TrackerRecHit2D/src/BaseTrackerRecHit.cc

```diff
@@ -1,3 +1,4 @@
+//#define DO_THROW_UNINITIALIZED
```

Comment:
shouldn't this be in .h file?
If define is added only here, then in all cases where the compiler decides to not inline the call of ```check``` where it's used, there will be a problem with one definition rule.